src/worker/core/GameSim.basketball.ts

```diff
@@ -2026,6 +2026,10 @@ class GameSim {
 
 	recordPlay(type: PlayType, t?: TeamNum, names?: string[], extra?: any) {
 		let texts;
+
+		const p = pickPlayer(this.playersOnCourt[this.d]);
```

Comment:
I'm not totally sure why this is failing so horribly, but it's still wrong. Look at other parts that do this, they're more like:

		const ratios = this.ratingArray("rebounding", this.o, 5);
		const p = this.playersOnCourt[this.o][pickPlayer(ratios)];

`ratios` is then the relative probability of being selected, like `[1, 1, 1, 1, 1]` would give equal odds to all players on the court.

That being said, you are right that even fixing this we lose the context of who fouled the player in an and-1. To handle that, I would...

1. Add a new argument to `recordPlay` to pass in who fouled the shooter.
2. If that is supplied, use that as `d` 99% of the time (or something).